src/components/CharacterDetails.js

```diff
@@ -0,0 +1,51 @@
+import { useHistory, useParams } from "react-router";
+import { useState, useEffect } from "react";
+
+export default function CharacterDetails() {
+  const { id } = useParams();
+  const history = useHistory();
+  const [character, setCharacter] = useState({});
+
+  useEffect(() => {
+    const url = `https://rickandmortyapi.com/api/character/${id}`;
+
+    fetch(url)
+      .then((res) => res.json())
+      .then((data) => {
+        setCharacter(data);
+      });
+  }, [id]);
+
+  return (
+    <div className="characterDetails">
+      <img
+        className="imgDetailCard"
+        src={character.image}
+        alt={character.image}
+      ></img>
+      <p>
+        <span className="titleDetail">Name</span>
+        <span className="textDetail">{character.name}</span>
+      </p>
+      <p>
+        <span className="titleDetail">Gender</span>
+        <span className="textDetail">{character.gender}</span>
+      </p>
+      <p>
+        <span className="titleDetail">Status</span>
+        <span className="textDetail">{character.status}</span>
+      </p>
+      <p>
+        <span className="titleDetail">Specie</span>
+        <span className="textDetail">{character.species}</span>
+      </p>
+      {character.type !== "" && (
+        <p>
+          <span className="titleDetail">Type</span>
+          <span className="textDetail">{character.type}</span>
+        </p>
+      )}
+      <button onClick={() => history.goBack()}>Back</button>
+    </div>
```

Comment:
I think this could use some more semantic changes.
I wouldn't expect the character info to be a paragraph. Yes, this is a bit nit-picky, but a list might have been more fitting
The `div` could be replaced with `article`